### PR TITLE
Fix warning with torch.meshgrid

### PR DESCRIPTION
**Pull request by pallgeuer**
## Motivation

`torch.meshgrid()` has started raising a warning when not called with an explicit indexing parameter:
https://pytorch.org/docs/stable/generated/torch.meshgrid.html

## Modification
Provide `indexing='ij'` to all calls to `torch.meshgrid` that don't already specify indexing.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/models/backbones/utils/transformer.py [hunk 1]
```diff
@@ -796,7 +796,8 @@ def gen_encoder_output_proposals(self, memory, memory_padding_mask,
                 torch.linspace(
                     0, H - 1, H, dtype=torch.float32, device=memory.device),
                 torch.linspace(
-                    0, W - 1, W, dtype=torch.float32, device=memory.device))
+                    0, W - 1, W, dtype=torch.float32, device=memory.device),
+                indexing='ij')
```
Comment (jin-s13) [inline mmpose/models/backbones/utils/transformer.py:800]: This function is torch.linspace instead of torch.meshgrid. Is this change intended?

**Comment by pallgeuer** [comment]
Refer to https://github.com/open-mmlab/mmdetection/pull/8090#issuecomment-1140937053 for discussion on PyTorch version support of `torch.meshgrid`.

File: mmpose/models/backbones/utils/transformer.py (hunk 1 above)
Comment (pallgeuer) [reply mmpose/models/backbones/utils/transformer.py:800]: If you expand the context you can see it is a call to `torch.meshgrid` with two calls to `torch.linspace` as arguments.
```
            grid_y, grid_x = torch.meshgrid(
                torch.linspace(
                    0, H - 1, H, dtype=torch.float32, device=memory.device),
                torch.linspace(
-                    0, W - 1, W, dtype=torch.float32, device=memory.device))
+                    0, W - 1, W, dtype=torch.float32, device=memory.device),
                indexing='ij')
```

Comment (jin-s13) [reply mmpose/models/backbones/utils/transformer.py:800]: Aha, my fault.

File: mmpose/models/detectors/multiview_pose.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def compute_grid(self, box_size, box_center, num_bins, device=None):
             grid_1D_x + box_center[0],
             grid_1D_y + box_center[1],
             grid_1D_z + box_center[2],
-        )
+            indexing='ij')
         grid_x = grid_x.contiguous().view(-1, 1)
```
Comment (ly015) [inline mmpose/models/detectors/multiview_pose.py:57]: The argument `indexing` has been introduced in PyTorch1.10.0 so we need to handle the compatibility of earlier versions. I think a version check could be added like:

```python
from packaging import version

if version.parse(torch.__verson__) < version.parse('1.10.0'):
    grid_x, grid_y, grid_z = torch.meshgrid(
            grid_1D_x + box_center[0],
            grid_1D_y + box_center[1],
            grid_1D_z + box_center[2])
else:
    grid_x, grid_y, grid_z = torch.meshgrid(
            grid_1D_x + box_center[0],
            grid_1D_y + box_center[1],
            grid_1D_z + box_center[2],
            indexing='ij')
```

**Comment by ly015** [comment]
Thank you very much for your help!

File: mmpose/models/detectors/multiview_pose.py (hunk 2 above)
Comment (pallgeuer) [reply mmpose/models/detectors/multiview_pose.py:57]: Yes, or seeing as there are multiple instances of `torch.meshgrid` one could write a wrapper for it that does the version check. Where would be an appropriate location for such a wrapper?

Comment (ly015) [reply mmpose/models/detectors/multiview_pose.py:57]: I think it's okay to directly use an if-else statement. If you prefer to use a wrapper, what about placing it at `mmpose/models/utils/misc.py`? Also, please note that `mmpose/models/backbones/utils/transformer.py` has been moved to 'mmpose/models/utils/transformer.py` this PR may need a rebase.

**Comment by pallgeuer** [comment]
I have updated the pull request to now correctly deal with PyTorch versions above and below 1.10.0 using a wrapper. Can the PR be reviewed if all is well now?

File: mmpose/models/detectors/multiview_pose.py (hunk 2 above)
Comment (pallgeuer) [reply mmpose/models/detectors/multiview_pose.py:56]: I have added a wrapper and adjusted

**Review by ly015 — APPROVED** [review]
